# **1- import libraries**

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px
import seaborn as sns

In [24]:
df = pd.read_csv('data.csv')
df

,price,brand,model,year,condition,kilo_meter,color,power_trans,fuel_type,location
0,"6,600,000",BMW,X6 M,2024,Used,"40,000 KM",Dark blue,Automatic,Gas,Tagamo3 - New Cairo
1,"5,600,000",Mercedes,GLC 200,2026,New,0 KM,Black,Automatic,Gas,Tagamo3 - New Cairo
2,"2,000,000",Cadillac,Escalade,2018,Used,"185,000 KM",Black,Automatic,Diesel,Tagamo3 - New Cairo
3,"2,000,000",Seat,Tarraco,2024,Used,"45,000 KM",Black,Automatic,Gas,Cairo
4,"3,650,000",Jeep,Grand Cherokee,2022,Used,"48,000 KM",White,Automatic,Gas,Cairo
...,...,...,...,...,...,...,...,...,...,...
24908,"EGP 1,385,000",Toyota,Corolla,2025,Used,"10,000",Gray,Automatic,Benzine,Giza
24909,"EGP 620,000",Kia,Rio,2017,Used,"135,000",Brown,Automatic,Benzine,Nasr City
24910,"EGP 1,795,000",Deepal,S07,2024,New,0,Gray,Automatic,Electric,Cairo
24911,"EGP 770,000",Chrysler,Town and Country,2010,Used,"300,000",Gray,Automatic,Benzine,Cairo


In [25]:
df.shape

(24913, 10)

In [27]:
df.head()

,price,brand,model,year,condition,kilo_meter,color,power_trans,fuel_type,location
0,"6,600,000",BMW,X6 M,2024,Used,"40,000 KM",Dark blue,Automatic,Gas,Tagamo3 - New Cairo
1,"5,600,000",Mercedes,GLC 200,2026,New,0 KM,Black,Automatic,Gas,Tagamo3 - New Cairo
2,"2,000,000",Cadillac,Escalade,2018,Used,"185,000 KM",Black,Automatic,Diesel,Tagamo3 - New Cairo
3,"2,000,000",Seat,Tarraco,2024,Used,"45,000 KM",Black,Automatic,Gas,Cairo
4,"3,650,000",Jeep,Grand Cherokee,2022,Used,"48,000 KM",White,Automatic,Gas,Cairo


In [28]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 24913 entries, 0 to 24912
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   price        24913 non-null  str  
 1   brand        24913 non-null  str  
 2   model        24913 non-null  str  
 3   year         24913 non-null  str  
 4   condition    24913 non-null  str  
 5   kilo_meter   24913 non-null  str  
 6   color        24913 non-null  str  
 7   power_trans  24913 non-null  str  
 8   fuel_type    24913 non-null  str  
 9   location     24913 non-null  str  
dtypes: str(10)
memory usage: 3.4 MB


--------> All columns are string data type

# **2- check missing & duplications**

In [29]:
df.isnull().sum()

price          0
brand          0
model          0
year           0
condition      0
kilo_meter     0
color          0
power_trans    0
fuel_type      0
location       0
dtype: int64

In [30]:
df.duplicated().sum()

np.int64(1014)

### --------> No Missing Value and find **1014** Duplicated rows

### --------> Delete duplicated data


In [31]:
df.drop_duplicates(inplace = True)

In [32]:
df.duplicated().sum()

np.int64(0)

# Check outliers

In [33]:
# Convert price column to numeric and Chick outliers
df['price'] = (
    df['price']
    .astype(str)
    .str.replace(',', '', regex=False)
    .str.replace('EGP', '', regex=False)
    .str.strip()
)

df['price'] = pd.to_numeric(df['price'], errors='coerce')

In [34]:
fig = px.box(
    df,
    x='price',
    title='Car Price Distribution After Removing Outliers'
)

fig.show()

In [35]:
df.to_csv('data.csv', index=False, encoding='utf-8-sig')